src/resmom/catch_child.c

```diff
@@ -340,29 +633,236 @@ encode_used(job *pjob, pbs_list_head *phead)
+								log_err(-1,
+									__func__,
+									log_buffer);
+								Py_CLEAR(py_jvalue);
+								fail = 1;
+							} else {
+								Py_CLEAR(py_jvalue);
+							}
+
+						} else  {
+							rd->rs_set(&tmpatr, &val2, INCR);
+						}
+						break;
+					}
+					if (fail) {
+						break;
+					}
+				}
+
+
+				if (val.at_type == ATR_TYPE_STR) {
+
+					if (fail) {
+						Py_CLEAR(py_accum);
+						/* unset resc */
+						(void)add_to_svrattrl_list(
+							phead,
+							ad->at_name,
+							rd->rs_name,
+							"",
+							SET, NULL);
+						/* go to next resource to encode_used */
+						continue;
+					}
+
+					sval = val.at_val.at_str;
+					if (PyDict_Size(py_accum) == 0) {
+						/* no other values seen
+						 * except from MS...use as is
+						 * don't JSONify */
+						rd->rs_decode(&tmpatr,
+							ATTR_used, rd->rs_name,
+							sval);
+						Py_CLEAR(py_accum);
+					} else if ((py_jvalue=json_loads(sval,
+						emsg, HOOK_BUF_SIZE-1)) == NULL) {
+						snprintf(log_buffer,
+							sizeof(log_buffer),
+							"Job %s resources_used.%s cannot be accumulated: value '%s' from mom %s not JSON-format: %s",
+							pjob->ji_qs.ji_jobid,
+							rd->rs_name, sval,
+							mom_short_name, emsg);
+						log_err(-1, __func__,
+							log_buffer);
+						Py_CLEAR(py_accum);
+						/* unset resc */
+						(void)add_to_svrattrl_list(
+							phead,
+							ad->at_name,
+							rd->rs_name,
+							"",
+							SET, NULL);
+						/* go to next resource to encode */
+						continue;
+					} else if (PyDict_Merge(py_accum, py_jvalue, 1) != 0) {
+						snprintf(log_buffer,
+							sizeof(log_buffer),
+							"Job %s resources_used.%s cannot be accumulated: value '%s' from mom %s: error merging values",
+							pjob->ji_qs.ji_jobid,
+							rd->rs_name, sval,
+							mom_short_name);
+						log_err(-1, __func__,
+							log_buffer);
+						Py_CLEAR(py_jvalue);
+						Py_CLEAR(py_accum);
+						/* unset resc */
+						(void)add_to_svrattrl_list(
+							phead,
+							ad->at_name,
+							rd->rs_name,
+							"",
+							SET, NULL);
+						/* go to next resource to encode */
+						continue;
+					} else {
+						char *dumps = NULL;
+
+						dumps = json_dumps(py_accum, emsg, HOOK_BUF_SIZE-1);
+						if (dumps == NULL) {
+							snprintf(log_buffer,
+								sizeof(log_buffer),
+								"Job %s resources_used.%s cannot be accumulated: %s",
+								pjob->ji_qs.ji_jobid,
+								rd->rs_name, emsg);
+							log_err(-1,
+								__func__,
+								log_buffer);
+							Py_CLEAR(py_jvalue);
+							Py_CLEAR(py_accum);
+							/* unset resc */
+							(void)add_to_svrattrl_list(
+								phead,
+								ad->at_name,
+								rd->rs_name,
+								"",
+								SET, NULL);
+							continue;
+						}
+
+						rd->rs_decode(&tmpatr, ATTR_used, rd->rs_name, dumps);
+						Py_CLEAR(py_jvalue);
+						Py_CLEAR(py_accum);
+						free(dumps);
+					}
+				}
+				val = tmpatr;
+#endif
```

Comment:
I thought about it but I feel that it's not necessary. Before this feature, this code has always been silent when encountering resources that mom does not accumulate.